src/io/flutter/run/daemon/DeviceService.java

```diff
@@ -205,10 +205,11 @@ private DeviceDaemon chooseNextDaemon(Refreshable.Request<DeviceDaemon> request)
       return previous; // Don't do anything; current daemon is what we want.
     }
 
-    // Wait a bit to see if we get cancelled.
-    // This is to try to avoid starting a process only to immediately kill it.
+    // Wait a bit to see if we get cancelled. This is to try to avoid starting a process only to
+    // immediately kill it. Also, delay a bit in case the flutter tool just upgraded the sdk;
+    // we'll need a bit more time to start up.
     try {
-      Thread.sleep(50);
+      Thread.sleep(2000);
```

Comment:
It's a good question; I'm going to think more about the timing of the sdk upgrades (this delay is in here to work around some exceptions I saw if we run the `flutter` tool too soon after the new sdk is downloaded - possibly a race condition with the flutter cli tool?).